status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import "sync"
+
+// Status represents a status update from an action within Empire.
+type Status interface {
```

Comment:
Yeah haha. If we wanted to tag on additional structured attributes, then an interface with different implementations would make sense. But for what we'll use this for right now, a struct is probably simpler.